src/compiler/scala/tools/nsc/backend/jvm/GenASM.scala

```diff
@@ -595,7 +595,8 @@ abstract class GenASM extends SubComponent with BytecodeWriters { self =>
           val x = innerClassSymbolFor(s)
           if(x ne NoSymbol) {
             assert(x.isClass, "not an inner-class symbol")
-            val isInner = !x.rawowner.isPackageClass
```

Comment:
No, the changes in [BCodeAsmCommon](https://github.com/scala/scala/pull/4264/files#diff-1) also apply to GenASM